pkg/k8sclient/k8sclient.go

```diff
@@ -35,37 +36,52 @@ func GetNodeByLabel(label string, kubeClient kubernetes.Interface) (v1.Node, err
 		for _, node := range nodeList.Items {
 			nodeCapacity, ok := node.Status.Capacity[LvCapacity]
 			if !ok {
-				return returnNode, errors.New("No lv-capacity set, yet!")
+				continue
 			}
 			if (&nodeCapacity).CmpInt64(maxCapacity) == 1 {
 				maxCapacity = (&nodeCapacity).Value()
 				returnNode = node
 			}
 		}
 	}
+	if returnNode.ObjectMeta.Name == "" {
+		return v1.Node{}, errors.New("No lv-capacity set, yet!")
+	}
 	return returnNode, nil
 }
 
 func UpdatePvc(pvc v1.PersistentVolumeClaim, kubeClient kubernetes.Interface) error {
-	_, err := kubeClient.CoreV1().PersistentVolumeClaims(pvc.ObjectMeta.Namespace).Update(&pvc)
+	_, err := kubeClient.CoreV1().PersistentVolumeClaims(pvc.ObjectMeta.Namespace).Update(context.TODO(), &pvc, metav1.UpdateOptions{})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 func GetNode(nodeName string, kubeClient kubernetes.Interface) (*v1.Node, error) {
-	node, err := kubeClient.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
+	node, err := kubeClient.CoreV1().Nodes().Get(context.TODO(), nodeName, metav1.GetOptions{})
  	if err != nil {
 		return nil, err
 	}
 	return node, nil
 }
 
 func UpdateNodeStatus(nodeName string, kubeClient kubernetes.Interface, node *v1.Node) error {
-	_, err := kubeClient.CoreV1().Nodes().UpdateStatus(node)
+	_, err := kubeClient.CoreV1().Nodes().UpdateStatus(context.TODO(), node, metav1.UpdateOptions{})
 	if err != nil {
 		return err
 	}
 	return nil
 }
+
+func StorageClassIsNokiaLocal(storageClassName string, kubeClient kubernetes.Interface) (bool, error){
```

Comment:
shoudnt this be added to the pvchandler entry criteria list too? after all we should do nothing on any level with a resource not belonging to us

Reply:
Pvcs have annotation "volume.beta.kubernetes.io/storage-provisioner" which is contains the provisoner of the storage-class. In our case it is "nokia.k8s.io/local". Pvs only have storagclass name.
Should I change to both check the storageclass name?

Reply:
would be good if the solution wouldn't depend on beta APIs IMO